File: sample-rxjava/src/test/java/org/dbtools/sample/model/database/MultiDatabaseManager.java
package org.dbtools.sample.model.database;

import org.dbtools.android.domain.AndroidDatabase;
import org.dbtools.android.domain.config.DatabaseConfig;

import javax.annotation.Nonnull;

public class MultiDatabaseManager extends DatabaseManager {
    public MultiDatabaseManager(DatabaseConfig databaseConfig) {
        super(databaseConfig);
    }

    @Override
    public void onCreate(@Nonnull AndroidDatabase androidDatabase) {
        createMainTables(androidDatabase);
    }

    @Override
    public void onCreateViews(@Nonnull AndroidDatabase androidDatabase) {
        createMainViews(androidDatabase);
    }

    @Override
    public void onDropViews(@Nonnull AndroidDatabase androidDatabase) {
        dropMainViews(androidDatabase);
    }
}

File: sample-rxjava-kotlin/src/test/kotlin/org/dbtools/sample/kotlin/model/database/MultiDatabaseManager.kt
package org.dbtools.sample.kotlin.model.database

import org.dbtools.android.domain.AndroidDatabase
import org.dbtools.android.domain.config.DatabaseConfig

class MultiDatabaseManager(databaseConfig: DatabaseConfig) : DatabaseManager(databaseConfig) {

    override fun onCreate(androidDatabase: AndroidDatabase) {
        createMainTables(androidDatabase)
    }

    override fun onCreateViews(androidDatabase: AndroidDatabase) {
        createMainViews(androidDatabase)
    }

    override fun onDropViews(androidDatabase: AndroidDatabase) {
        dropMainViews(androidDatabase)
    }
}
